custom_model_runner/datarobot_drum/drum/gpu_predictors/nemo_predictor.py

```diff
@@ -0,0 +1,239 @@
+"""
+Copyright 2024 DataRobot, Inc. and its affiliates.
+All rights reserved.
+This is proprietary source code of DataRobot, Inc. and its affiliates.
+Released under the terms of DataRobot Tool and Utility Agreement.
+"""
+import csv
+import io
+import logging
+import os
+import signal
+import subprocess
+import atexit
+
+import numpy as np
+from openai import OpenAI
+from threading import Thread
+from datarobot_drum import RuntimeParameters
+from datarobot_drum.drum.adapters.model_adapters.python_model_adapter import RawPredictResponse
+from datarobot_drum.drum.common import SupportedPayloadFormats
+from datarobot_drum.drum.enum import (
+    LOGGER_NAME_PREFIX,
+    PayloadFormat,
+    StructuredDtoKeys,
+)
+from datarobot_drum.drum.exceptions import DrumCommonException
+from datarobot_drum.drum.language_predictors.base_language_predictor import BaseLanguagePredictor
+from datarobot_drum.resource.drum_server_utils import wait_for_server, DrumServerProcess
+
+RUNNING_LANG_MSG = "Running environment: Nemo Inference Microservices."
+DEFAULT_MODEL_NAME = "generic_llm"
+
+
+class ChatRoles:
+    SYSTEM = "system"
+    ASSISTANT = "assistant"
+    USER = "user"
+
+
+class NemoPredictor(BaseLanguagePredictor):
+    def __init__(self):
+        super(NemoPredictor, self).__init__()
+        self.logger = logging.getLogger(LOGGER_NAME_PREFIX + "." + __name__)
+        self.datarobot_venv_path = os.environ.get("DATAROBOT_VENV_PATH", "/home/nemo/dr")
+        # expected to be set for text generation type
+        self.prompt_field = os.environ.get("TARGET_NAME")
```

Comment:
From my experience, this isn't the prompt field, this is the response field. Buzok appears to use a runtime param to define the prompt field.

Reply:
ok, buzok has the `TARGET_NAME` and the `PROMPT_COLUMN_NAME`. but why, isn't that confusing?
![buzok-model](https://github.com/datarobot/datarobot-user-models/assets/173610/82dfbe12-cb79-4e83-9dcf-e341fa84c117)

Reply:
![text-gen-model-target-name](https://github.com/datarobot/datarobot-user-models/assets/173610/764b72ce-071f-4a60-b075-02a675a0dbe1)

Reply:
the `TARGET_NAME` env var is set from this UI field. What Buzok does is a question :question:

Reply:
updated code to read a prompt filed name from the `MLOPS_RUNTIME_PARAM_prompt_column_name`. 

Buzok uses upper case for parameter names, I chose to use lowercase to be similar to [OpenAI request parameters](https://platform.openai.com/docs/api-reference/chat/create).